migrator/runner/runner.go

```diff
@@ -84,11 +99,17 @@ func runMigrations(databases *types.Databases, startingSeqNum int) error {
 		} else {
 			err := migration.Run(databases)
 			if err != nil {
-				return errors.Wrapf(err, "error running migration starting at %d", seqNum)
+				return wrapRollback(ctx, tx, errors.Wrapf(err, "error running migration starting at %d", seqNum))
 			}
 		}
 
-		err := updateVersion(databases, migration.VersionAfter)
+		err = tx.Commit(ctx)
+		if err != nil {
+			return wrapRollback(ctx, tx, errors.Wrapf(err, "unable to commit migration starting at %d", seqNum))
+		}
+
+		// Outside transaction because of Gorm of course
+		err = updateVersion(databases, migration.VersionAfter)
```

Comment:
Can we put this transaction together with the migration? Otherwise we should handle the case of re-entry here.